### DISCUSSION TITLE
createForm创建时机？

### DISCUSSION THREAD

**Discussion by 729993031** [alibaba/formily#1782] [Q&A 2.x]
我看到https://react.formilyjs.org/api/components/field 这个代码示例的createForm是在函数外面的 是否正确？

import React from 'react'
import { createForm } from '@formily/core'
import { FormProvider, Field } from '@formily/react'
import { Input } from 'antd'

const form = createForm()

export default () => (
  <FormProvider form={form}>
    <Field name="input" component={[Input, { placeholder: 'Please Input' }]} />
  </FormProvider>
)

**Comment by liuweiGL** [comment]
你可能需要先去理解 form 的作用是什么，再来思考这个问题。

**Reply by 729993031** [reply]
看到了这个讨论 https://github.com/alibaba/formily/discussions/1223 
没想到是FormProvider做了这个回收 我原本的担心是生命周期结束后不会对这个创建的form进行清除
